packages/components/src/templates/table.stories.ts

```diff
@@ -0,0 +1,596 @@
+                })}
+              </tr>`;
+            })()}
+          </tfoot>
+        </table>
+
+        <script>
+          document.addEventListener('DOMContentLoaded', () => {
+            const scrollableTable = document.getElementById('horizontal-scrollable-table');
+            const tableHeaders = scrollableTable.querySelectorAll('th');
+
+            scrollableTable.addEventListener('scroll', () => {
+              tableHeaders.forEach(header => {
+                if (scrollableTable.scrollLeft === 0) {
+                  header.classList.remove('sd-table-cell--shadow-active');
+                } else {
+                  header.classList.add('sd-table-cell--shadow-active');
+                }
+              });
+            });
+          });
+        </script>
+        <div class="headline">Fixed, vertical, left header with shadow table</div>
+        <div id="horizontal-scrollable-table" class="overflow-x-scroll overflow-y-visible w-[600px]">
+          <table class="sd-table sample-table">
+            ${tableData.map((rowData, rowIndex) => {
+              return html`<tr>
+                ${rowData.map((cellData, columIndex) => {
+                  if (columIndex === 0) {
+                    return html`<th
+                      class="sd-table-cell top-auto sticky left-0 z-[2] bg-clip-padding sd-table-cell--shadow-right ${rowIndex %
+                        2 ===
+                      0
+                        ? 'sd-table-cell--bg-white'
+                        : 'sd-table-cell--bg-primary-100'}"
+                    >
+                      Header
+                    </th>`;
+                  } else {
+                    return html`<td
+                      class="sd-table-cell text-nowrap whitespace-nowrap ${rowIndex % 2 === 0
+                        ? 'sd-table-cell--bg-white'
+                        : 'sd-table-cell--bg-primary-100'}"
+                    >
+                      ${cellData}
+                    </td>`;
+                  }
+                })}
+              </tr>`;
+            })}
+          </table>
+        </div>
+
+        <script>
+          document.addEventListener('DOMContentLoaded', () => {
+            const scrollableTable = document.getElementById('horizontal-scrollable-table-2');
+            const tableHeaders = scrollableTable.querySelectorAll('th');
+            const maxScrollX = scrollableTable.scrollWidth - scrollableTable.clientWidth;
+
+            scrollableTable.addEventListener('scroll', () => {
+              tableHeaders.forEach(header => {
+                if (scrollableTable.scrollLeft === maxScrollX) {
+                  header.classList.remove('sd-table-cell--shadow-active');
+                } else {
+                  header.classList.add('sd-table-cell--shadow-active');
+                }
+              });
+            });
+          });
+        </script>
+        <div class="headline">Fixed, vertical, right header with shadow table</div>
```

Comment:
for this advanced sample I would like to show neutral-100 instead of primary-100 so that we also show this color option

Reply:
updated.